### PR TITLE
Upload packages to s3 instead of local file system

### PR DESCRIPTION
**Pull request by simao**
Support local storage when no s3 credentials are available

Return 302/Found when client requests package stored on s3

### CODE REVIEW

File: core/src/main/scala/org/genivi/sota/core/PackagesResource.scala [hunk 1]
```diff
@@ -112,7 +77,7 @@ class PackagesResource(resolver: ExternalResolverClient, db : Database)
         completeOrRecoverWith(
           for {
             _ <- resolver.putPackage(ns, pid, description, vendor)
-            (uri, size, digest) <- savePackage(pid, fileData)
+            (uri, size, digest) <- packageStorage.store(pid, fileData)
```
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/PackagesResource.scala]: Can this be enabled by config? Default local storage.

**Comment by jerrytrieu** [comment]
Does rvi/TransferController need to be updated?

File: core/src/main/scala/org/genivi/sota/core/PackagesResource.scala (hunk 1 above)
Comment (simao) [inline core/src/main/scala/org/genivi/sota/core/PackagesResource.scala]: yeah `packageStore.store` will save it either to s3 or local fs, depending on config.

**Comment by simao** [comment]
Did not think about that... I think it needs to be updated to store the file in s3 after it finishes receiving it, and updating the db with the new s3 url... :|

File: core/src/main/scala/org/genivi/sota/core/PackagesResource.scala (hunk 1 above)
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/PackagesResource.scala]: cool

**Comment by jerrytrieu** [comment]
We only download over rvi, only need to read. No need to update url.

**Comment by simao** [comment]
ok that is more complicated than I thought then, because we need to download the file from s3 before starting to upload the file to the client :|

**Comment by jerrytrieu** [comment]
If simpler, we can enable s3 only if RVI is disabled. TODO later.

**Comment by simao** [comment]
Implemented pre downloading of the s3 file on when using rvi.

the tests are red because of some error with docker compose. Did something change??

**Pull request merged**